aiutils/data/data.py

```diff
@@ -0,0 +1,87 @@
+"""Utility functions for making batch generation easier."""
+import numpy as np
+from multiprocessing import Process, Queue
+
+
+def sequential(batch_size, num_samples, num_epochs=1):
+    """Generate sequence indices.
+    """
+    for epoch in range(num_epochs):
+        indices = np.arange(num_samples)
+        for i in range(0, num_samples - batch_size + 1, batch_size):
+            slice_ = slice(i, i + batch_size)
+            yield indices[slice_]
+
+
+def random(batch_size, num_samples, num_epochs=1):
+    """Generate random indices.
+    """
```

Comment:
You mean instead of random?

So I agree that would work, but then you have to decide when to shuffle, and how many times to shuffle. I don't see why you should re-write that each time, when all you want to do is grab random batches.